### DISCUSSION TITLE
rich.pretty.pprint should have the option to sort unordered collections like set() and dict()

### DISCUSSION THREAD

**Discussion by zachriggle** [Textualize/rich#1651] [Ideas] [answered]
`rich.pretty.pprint` is one of my favorite things, especially when in the reply with `rich.pretty.install()`!

One thing that would be nice is if unordered collections (like `set()` and `dict()` and things that subclass them) could be pretty-printed such that they are sorted by key (`dict`) or value (`set`).

For compatibility reasons (and since Python3 `dict` objects are technically ordered by insertion order), this would probably be behind a flag for `pprint` and `pretty.install`.

The Feature Request template suggested I discuss this here first, so here we are.

**Comment by willmcgugan** [comment, marked as answer]
Sorting containers would be reasonable. I agree it should be behind a flag, and disabled by default. Feel free to file a feature request issue,
